### PR TITLE
PEOPLE-101 unpend and fix projects and filters specs

### PR DESCRIPTION
**Pull request by mczuchnowski**
Complete rewrite of projects_spec and filters_spec

**Changes:**
- unpended pending project and filter specs
- refactored specs that did not work with the new app
- removed specs that check features that no longer exist

**Jira:**
https://netguru.atlassian.net/browse/PEOPLE-101

This is part 1 of a bigger effort to bring back feature specs to People after the app was rewritten in React.

### CODE REVIEW

File: spec/features/projects_spec.rb [hunk 1]
```diff
@@ -253,8 +239,7 @@
         expect(page).to have_selector('div.note-group')
         expect(page).to have_content(note.text)
 
-        # HACK: because capybara has problems clicking links with empty body
-        execute_script "$('.note-remove').click()"
+        find('.note-remove').click()
```
Comment (netguru-hound) [inline spec/features/projects_spec.rb]: Do not use parentheses for method calls with no arguments.

File: spec/features/projects_spec.rb [hunk 2]
```diff
@@ -190,39 +179,36 @@
...
 
         billable_count = find('.billable .count')
         expect(billable_count).to have_content('1')
       end
     end
 
-    describe 'removing member from project' do
-      let!(:membership) { create(:membership, user: pm_user, project: active_project) }
+    describe 'ending membership in a regular project' do
+      let!(:membership) { create(:membership, user: admin_user, project: active_project, ends_at: nil) }
```
Comment (netguru-hound) [inline spec/features/projects_spec.rb]: Line is too long. [104/100]

File: spec/features/filters_spec.rb [hunk 3]
```diff
@@ -40,36 +41,32 @@
     context 'when user has not started a project' do
       let!(:junior_role) { create(:role, name: 'junior') }
       let!(:future_dev) { create(:user, primary_role: junior_role) }
-      let!(:future_membership) { create(:membership, user: future_dev, starts_at: 1.week.from_now) }
+      let!(:future_membership) { create(:membership, user: future_dev, starts_at: 1.week.from_now,
```
Comment (netguru-hound) [inline spec/features/filters_spec.rb]: Avoid using `{...}` for multi-line blocks.<br>Block body expression is on the same line as the block start.

File: spec/features/filters_spec.rb [hunk 4]
```diff
@@ -40,36 +41,32 @@
...
+        project: project_zztop) }
```
Comment (netguru-hound) [inline spec/features/filters_spec.rb]: Align the elements of a hash literal if they span more than one line.<br>Expression at 45, 33 should be on its own line.

File: spec/features/filters_spec.rb [hunk 5]
```diff
@@ -40,36 +41,33 @@
     context 'when user has not started a project' do
       let!(:junior_role) { create(:role, name: 'junior') }
       let!(:future_dev) { create(:user, primary_role: junior_role) }
-      let!(:future_membership) { create(:membership, user: future_dev, starts_at: 1.week.from_now) }
+      let!(:future_membership) do
+        create(:membership, user: future_dev, starts_at: 1.week.from_now, project: project_zztop)
```
Comment (mateuszgorniak) [inline spec/features/filters_spec.rb]: what about creating extra trait for future membership?

Comment (mczuchnowski) [inline spec/features/filters_spec.rb]: That is actually a great suggestion! I was too focused on making these specs at least work again ;)

**Comment by mateuszgorniak** [comment]
Review done, @mczuchnowski :-)

**Pull request merged**